src/sentry/utils/transitional_serializer.py

```diff
@@ -11,9 +12,11 @@ def __init__(self) -> None:
         self.json_serializer = JSONSerializer()
 
     def dumps(self, obj: Dict[str, Any]) -> bytes:
-        # will have the write as pickle obj for now, will change to json in seperate deploy
-        # return self.json_serializer.dumps(obj)
-        return self.pickle_serializer.dumps(obj)
+        for key in obj:
```

Comment:
what if the timestamp is in a nested part of the object?

Reply:
how will this work when loading? don't we need changes for that?

Reply:
good point on the nested part, ill come up with a way on how to do that. using the same logic, i can implement a way to load it back to a datetime obj using the same logic in the loads fn

Reply:
okay. an alternative would be to convert the callers to pass in a raw timestamp to avoid having to convert it here -- i'll let you decide.

Reply:
the way i thought of dealing with a nested datetime obj is to recursively call a helper fn if a type dict is in the value.  then once it loads it calls another helper fn to go thru with the same logic of converting datetime to timestamp but the other way around to turn it back. i've pushed up the logic, lmk if its good or a more efficient method should be used